app/controllers/miq_action_controller.rb

```diff
@@ -97,7 +97,7 @@ def action_field_changed
       update_playbook_variables(params)
     end
     @edit[:new][:options][:service_template_id] = params[:service_template_id].to_i if params[:service_template_id]
-
+    @edit[:new][:options][:service_template_name] = ServiceTemplateAnsiblePlaybook.order(:name).pluck(:name, :id)
```

Comment:
Could you id in line no 99 to get service template name? then , you can just call @record.options[:service_template_name] in haml, then, no need to add any extra logic in haml.